### PR TITLE
[BUD-223] Feature - Delete Task

### PR DESCRIPTION
**Pull request by Tonik1204**
# [#223](https://digitalrig.atlassian.net/browse/BUD-223) Feature / Task - delete

## Description

- Add Figma wireframes [here](https://www.figma.com/proto/lw9LmbrTtHA1LkcmDx2T0z/Buddy-app-interactive-prototype?node-id=2595%3A834&scaling=min-zoom);
- Add Task Options component;
- Split Task Checkbox component into Task List Item and Task Checkbox components;
- Update delete task logic on server side + schema;
- Udpate AlertDialog to use Context API;
- Add unit tests;

# Screenshots

ezgif com-video-to-gif (1)
ezgif com-video-to-gif

### CODE REVIEW

File: src/components/TaskOptions/TaskOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        try {
+          deleteTask && (await deleteTask({ variables: { taskId, newbieId } }));
+          showSnackbar(DICTIONARY.DELETE_SNACKBAR.SUCCESS);
+        } catch (err) {
+          showSnackbar(DICTIONARY.DELETE_SNACKBAR.ERROR);
+        }
+      }
+    );
+
+  const options = [
```
Comment (michaelmoney) [inline src/components/TaskOptions/TaskOptions.tsx]: Maybe it's a good idea to define types for the option object.

File: schema/src/generated/types.ts [hunk 2]
```diff
@@ -63,7 +63,7 @@ export type Mutation = {
   login: AuthPayload;
   addNewbieTask: Task;
   addBuddyTask: Task;
-  deleteTask: Task;
+  deleteTask: Newbie;
```
Comment (lucianojss) [inline schema/src/generated/types.ts]: Why `deletedTask` is now a `Newbie` type?

File: schema/src/schema.graphql [hunk 3]
```diff
@@ -30,7 +30,7 @@ type Mutation {
   login(email: String!, password: String!): AuthPayload!
   addNewbieTask(newbieId: ID!, input: TaskInput!): Task!
   addBuddyTask(newbieId: ID!, input: TaskInput!): Task!
-  deleteTask(taskId: ID!): Task!
+  deleteTask(taskId: ID!, newbieId: ID!): Newbie!
```
Comment (lucianojss) [inline schema/src/schema.graphql]: Isn't it just `TaskId` sufficient to remove a task?
Why removing a task returns a Newbie? 🤔

File: schema/src/generated/types.ts (hunk 2 above)
Comment (Tonik1204) [reply schema/src/generated/types.ts]: We need `Newbie` to auto update apollo cache

File: schema/src/schema.graphql (hunk 3 above)
Comment (Tonik1204) [reply schema/src/schema.graphql]: Answered [here](https://github.com/BuildIt-Poland/buddy-app/pull/53#discussion_r363790338)

File: src/components/TaskOptions/TaskOptions.tsx (hunk 1 above)
Comment (Tonik1204) [reply src/components/TaskOptions/TaskOptions.tsx]: Fixed here [#6f88e07](https://github.com/BuildIt-Poland/buddy-app/pull/53/commits/6f88e078eb6ab5479d1ffc702b6e5ceea736b954#diff-0fd0bc4acdcd9be9127919bea5cd012c)

File: src/components/AlertDialog/AlertDialog.tsx [hunk 4]
```diff
@@ -1,40 +1,44 @@
...
-}
-
-const AlertDialog: React.FC<AlertDialogProps> = ({ title, message }) => {
-  const [open, setOpen] = React.useState(true);
-
-  const handleClose = () => {
-    setOpen(false);
-  };
+const AlertDialog: React.FC = () => {
+  const { isOpen, title, message, onConfirm, hideDialog } = useContext<
```
[the remark this answers is not in the document]
Comment (Tonik1204) [reply src/components/AlertDialog/AlertDialog.tsx]: I guess, better to keep it in your PR related to context changes

File: src/__mocks__/context.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  isAuthenticated: true,
+  loading: false,
+  data: {
+    role: UserRole.Buddy,
+    token: 'token',
+    userId: '1234',
+  },
+};
+
+export const mockedNewbieContext: AuthState = {
```
[the remark this answers is not in the document]
Comment (Tonik1204) [reply src/__mocks__/context.ts]: Got it, ok so now it's a function with possibility to add partialy changes to context data

File: src/components/App/App.tsx [hunk 6]
```diff
@@ -24,6 +25,7 @@ const App: React.FC = () => {
         fallback={<PageContainer data-testid='loader' backGroundShape />}>
         {isAuthenticated ? <AuthenticatedApp /> : <NotAuthenticatedApp />}
       </React.Suspense>
+      <AlertDialog />
```
[the remark this answers is not in the document]
Comment (Tonik1204) [reply src/components/App/App.tsx]: I guess, better to keep it in your PR related to context changes

**Pull request merged**